### PR TITLE
Fixed constant loss and accuracy issue

### LINKED ISSUE #72: Fix constant loss and accuracy
The loss should generally be decreasing and the accuracy should generally be decreasing.

### PR DESCRIPTION
**Pull request by dscolby**
## Describe your changes
The loss now decreases and the accuracy increases instead of staying constant when training. Fixes #72 and #75.

## Checklist before merging

- [x ] My code produces no errors when run
- [ x] My code is linted and properly formatted

### CODE REVIEW

**Comment by dscolby** [comment]
These updates close #72 and #75.

File: backend/ml/ml/train.py [hunk 1]
```diff
@@ -1,111 +1,137 @@
...
     Returns:
     --------
-    
+        A tuple of updated parameters, loss, gradients, and accuracy
     """
-    # Binary cross entropy with clipping to avoid rounding issues
+
     def loss_fn(params, x, y):
         ys = foward_deep_fm(params, x)
         ys = jnp.clip(ys, 1e-7, 1 - 1e-7)
```
Comment (chanteriam) [inline backend/ml/ml/train.py:61]: It may be nice to save 1e-7 to a const to make it more clear what it is used for

File: backend/ml/ml/train.py [hunk 2]
```diff
@@ -1,111 +1,137 @@
...
 
     Returns:
     --------
-        tuple (list): a list of epochs, a list of the loss at each epoch, and a list 
-            of the accuracy at each epoch
+        A tuple containing lists of epochs, loss, accuracy, and parameters
     """
     epochs, loss_list, acc_list = [], [], []
-
     solver = optax.adam(0.0001)
```
Comment (chanteriam) [inline backend/ml/ml/train.py:86]: same here re: const for 0.0001

**Review by chanteriam — APPROVED** [review]

File: backend/shoulder/ml/tests/test_train.py [hunk 3]
```diff
@@ -1,21 +1,70 @@
...
 from ml.model import init_deep_fm
 from pathlib import Path
+>>>>>>> 03e10757a2d533e24b95eb4b5551f20a6dc8cf1e
 
 
 def test_step():
     x_train = random.randint(random.PRNGKey(97), shape=(1000, 5), minval=0, 
                              maxval=50).astype(float)
     y_train = random.bernoulli(random.PRNGKey(997), 0.35, shape=(1000,)).astype(float)
+<<<<<<< HEAD
```
Comment (ethanarsht) [inline backend/shoulder/ml/tests/test_train.py:23]: Probably don't want this version control changelog

**Review by ethanarsht — DISMISSED** [review]
Did you mean to merge this into main? Seems like it's headed for dev

**Comment by chanteriam** [comment]
> Did you mean to merge this into main? Seems like it's headed for dev

No talk changes should be merged to development right now.

**Review by chanteriam — DISMISSED** [review]
